api/api/auth/tests.py

```diff
@@ -99,7 +99,7 @@ def test_createToken_userExists_tokenDoesNotExist_createsTokenAndReturnsWithUser
     @mock.patch('time.sleep')
     @mock.patch('google_auth_oauthlib.flow.Flow.fetch_token')
     @mock.patch('google.oauth2.id_token.verify_oauth2_token')
-    def test_createToken_tokenExists_returnsExistingTokenWithUserInfo(
+    def test_createNewToken_tokenExists_returnsNewTokenWithUserInfo(
```

Comment:
Try to not edit existing tests if new they are needed. Add a new test for your funcionality.

Reply:
> Try to not edit existing tests if new they are needed. Add a new test for your functionality.

This tests gets its response from `/auth/token` and the `post` method in `TokenView` uses different function which return a `new token` so this test will always fail

Reply:
> You modified the `find_or_create` method. Which now always returns a new token regardless of whether it's expired or not.
> 
> The test would pass if the `find_or_create` method stays the same. Figure out a way to remove the expired token and create a new one only if it's expired.
> 
> No need to generate a new token every time TokenView is called.

Probably I am confused with the implementation. I did not see any criteria to validate whether the token is expired or not .I think we need to set some?

Reply:
There's a `created_at` field in the `Token` model. Use that to check if it's expired.

Token is expired after 1 day.

Reply:
> There's a `created_at` field in the `Token` model. Use that to check if it's expired.
> 
> Token is expired after 1 day.

Sir we need to modify  this `def test_createToken_tokenExists_returnsExistingTokenWithUserInfo` coz it does not satisfy our way of implementation. It should be `def test_createToken_validTokenExists_returnsExistingTokenWithUserInfo`